com.civicrm.hrjobroles/tests/phpunit/api/v3/ContactHRJobRoleTest.php

```diff
@@ -99,9 +99,9 @@ public function testTheGetActionReturnsMultipleJobRoles() {
     $contact2 = ContactFabricator::fabricate();
     $contract1 = HRJobContractFabricator::fabricate(['contact_id' => $contact1['id']]);
     $contract2 = HRJobContractFabricator::fabricate(['contact_id' => $contact2['id']]);
-    $jobRole1 = HRJobRolesFabricator::fabricate(['job_contract_id' => $contract1['id'],]);
-    $jobRole2 = HRJobRolesFabricator::fabricate(['job_contract_id' => $contract1['id'],]);
-    $jobRole3 = HRJobRolesFabricator::fabricate(['job_contract_id' => $contract2['id'],]);
+    $jobRole1 = HRJobRolesFabricator::fabricate(['job_contract_id' => $contract1['id'], 'start_date' => date('Y-m-d H:i:s'),]);
```

Comment:
It will be better to make start_date with a default value of current date here for the fabricator here: https://github.com/compucorp/civihr/blob/43f80260a8aa664afe5b943fee4878003fc2e208/com.civicrm.hrjobroles/CRM/Hrjobroles/Test/Fabricator/HrJobRoles.php#L6, that way, there will be no need to modify any tests that use the fabricator and it won't break tests. If a test needs a different date, it can be set explicitly.

Reply:
Expression will not be allowed as field default value. Can you suggest an alternative implementation 🤷‍♂️  ?

Reply:
What about modifying the fabricate method like this?
```php
  public static function fabricate($params) {
    //Setting the default start date here when not provided
    if (empty($params['start_date'])) {
      $params['start_date'] = date('Y-m-d H:i:s');
    }
    $result = civicrm_api3(
      'HrJobRoles',
      'create',
      array_merge(self::$defaultParams, $params)
    );

    return array_shift($result['values']);
  }
```